src/middlewared/middlewared/plugins/docker/validation_utils.py

```diff
@@ -15,7 +15,11 @@ def validate_address_pools(system_ips: list[dict], user_specified_networks: list
     ])
     seen_networks = set()
     for index, user_network in enumerate(user_specified_networks):
-        base_network = ipaddress.ip_network(user_network['base'], False)
+        if isinstance(user_network['base'], ipaddress.IPv4Interface):
```

Comment:
You don't need to do this check. The `ipaddress` modules accept an instance of themselves as an argument. I discovered this when I made changes converting the static route plugin (See [here](https://github.com/truenas/middleware/pull/14759/files#diff-c763d4b1445b671af3b54cbf8038617f8705014f593f4cda024cbd4286734d12R116))

Reply:
This is for the unit tests i have for the same endpoint, hence the explicit check